src/txn/txn_rollback_to_stable.c

```diff
@@ -1069,6 +1077,13 @@ __rollback_to_stable_btree_hs_truncate(WT_SESSION_IMPL *session, uint32_t btree_
 
     for (; ret == 0; ret = hs_cursor->next(hs_cursor)) {
         WT_ERR(hs_cursor->get_key(hs_cursor, &hs_btree_id, hs_key, &hs_start_ts, &hs_counter));
+        /*
+         * If the stop time pair on the tombstone in the history store is already globally visible
+         * we can skip it.
+         */
+        if (__wt_txn_visible_all(
+              session, cbt->upd_value->tw.stop_txn, cbt->upd_value->tw.durable_stop_ts))
+            continue;
```

Comment:
This is the only visible check that happens before the btree ID check. Was that ordering difference deliberate here? Otherwise can this go after line 1090?